cmd/cmd.go

```diff
@@ -78,6 +92,22 @@ func ExecInCoreSnap() {
 		}
 	}
 
+	// ensure we do not re-exec into an older version of snapd
+	currentSnapd, err := os.Stat(exe)
+	if err != nil {
+		logger.Noticef("cannot stat %q: %s", exe, err)
+		return
+	}
+	coreSnapSnapd, err := os.Stat(full)
+	if err != nil {
+		logger.Noticef("cannot stat %q: %s", full, err)
+		return
+	}
+	if currentSnapd.ModTime().After(coreSnapSnapd.ModTime()) {
```

Comment:
As discussed this will be improved and we will use a /usr/lib/snapd/info file with "VERSION=$VER" in it.